examples/connectivity/plot_extract_regions_canica_maps.py

```diff
@@ -0,0 +1,84 @@
+"""
+Regions extraction using Canonical ICA maps and functional connectomes
+======================================================================
+
+This example shows how to use :class:`nilearn.regions.region_extractor.RegionExtractor`
+to extract connected brain regions from whole brain ICA maps and
+use them to estimate a connectome.
+
+We used 20 resting state ADHD functional datasets from :func:`nilearn.datasets.fetch_adhd`
+and :class:`nilearn.decomposition.canica` for whole brain ICA maps.
+
+Please see the related documentation of :class:`nilearn.regions.region_extractor.RegionExtractor`
+for more details.
+"""
+import numpy as np
+from nilearn import datasets
+print(" -- Fetching ADHD resting state functional datasets -- ")
+adhd_dataset = datasets.fetch_adhd(n_subjects=20)
+func_filenames = adhd_dataset.func
+confounds = adhd_dataset.confounds
+
+from nilearn.decomposition.canica import CanICA
+print(" -- Canonical ICA decomposition of functional datasets -- ")
+# Initialize canica parameters
+n_components = 5
+canica = CanICA(n_components=n_components, smoothing_fwhm=6.,
+                memory="nilearn_cache", memory_level=2,
+                threshold=3., random_state=0)
+
+canica.fit(func_filenames)
+components_img = canica.masker_.inverse_transform(canica.components_)
+
+from nilearn.regions import region_extractor
+print(" -- Extracting regions from ICA maps and timeseries signals -- ")
+extractor = region_extractor.RegionExtractor(
+    components_img, threshold="98%", standardize=True, min_size=100)
```

Comment:
Following our discussion, we have now 3 thresholding strategies:
* `img_value`: it's a float
* `percentile`: it can be a string, or be a float
* `ratio_n_voxel`: it's a float

So, if you remove `threshold_strategy`, we won't be able to differentiate between cases 1 and 3. I think you should just leave it as-is.